cmd/ingester/app/consumer/consumer.go

```diff
@@ -52,12 +56,15 @@ type consumerState struct {
 
 // New is a constructor for a Consumer
 func New(params Params) (*Consumer, error) {
+	deadlockDetectorFactory := newDeadlockDetectorFactory(params.Factory, params.Logger, time.Minute)
```

Comment:
separate issue: s/params.Factory/params.MetricsFactory/

Reply:
I'll address this separately - it'll only add noise to this PR